### ISSUE TITLE
Navigating in admin results in white page

### ISSUE THREAD

**Issue by Renkas** [strapi/strapi#16998]
## Bug report
When navigating in the admin panel every now and then it results to a white page. With following error in the console:
```
main.6c267524.js:2168 Uncaught TypeError: J is not a function
    at bc (main.6c267524.js:2168:23372)
    at Za (main.6c267524.js:2168:25181)
    at Mh (main.6c267524.js:2168:43442)
    at Ef (main.6c267524.js:2168:42970)
    at Ah (main.6c267524.js:2168:41987)
    at uh (main.6c267524.js:2168:38457)
    at vl (main.6c267524.js:2166:3311)
    at main.6c267524.js:2168:35751
```

This started in 4.11.0
Yesterday I updated to 4.11.1 and it still persists.

I have disabled all the plugins. But this has not fixed the issue.

### Required System information

- Node.js version: 18.16.0
- NPM version: 9.5.1
- Strapi version: 4.11.1
- Database: N/A
- Operating system: N/A
- Is your project Javascript or Typescript: TypeScript

**Comment by Kyle772** [comment]
+1 I'm getting this error after MOST page navigations after updating 4.11 and 4.11.1 also running typescript

**Comment by nitinmadelyn** [comment]
@Kyle772 @Renkas if you can post screen recording for this error then it would be helpful. Coz I have installed a fresh quickstart project using Typescript and can't see an error like this.

**Comment by Renkas** [comment]
@nitinmadelyn you click on a link in menu and the page turns to white (50% of time, it does not matter which menu item). Nothing much to record.

**Comment by Renkas** [comment]
Maybe it has something to do with what dependencies are installed?

But this problem persists even when /config/plugins.ts is empty - so these dependencies should not be loaded I believe?
```
export default ({env}) => ({
});
```

```
  "dependencies": {
    "@_sh/strapi-plugin-ckeditor": "^2.0.4",
    "@strapi/plugin-i18n": "^4.10.6",
    "@strapi/plugin-seo": "^1.9.1",
    "@strapi/plugin-users-permissions": "^4.10.6",
    "@strapi/provider-upload-aws-s3": "^4.10.6",
    "@strapi/strapi": "^4.10.6",
    "mysql": "2.18.1",
    "strapi-plugin-populate-deep": "^2.0.0",
    "strapi-plugin-sitemap": "^2.0.10",
    "strapi-plugin-transformer": "^3.1.0"
  },
```

**Comment by Schlipak** [comment]
@nitinmadelyn the settings page works, I only get the white page issue in _some of_ the content manager pages, and it seems to throw when the content editor (the rightmost pane) itself loads.
I can also confirm that a fresh Strapi project with Typescript works as intended.

https://github.com/strapi/strapi/assets/8151004/b527cd64-aab5-4731-a651-55126901bfef

Note that the "project" content-type does not crash, but uses custom fields, components and dynamic zones. After recording that video, I added the SEO component to the project content-type, and while editing a project does not cause an error, exiting the editor by going back to the projects list or by selecting another content-type now crashes the rightmost pane.

The console shows the following error while editing a project with the newly added SEO component:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'seoChecks')
    at $r (main.8aa1330e.js:1641:16112)
    at async main.8aa1330e.js:1641:17374
```

@Renkas Upgrading all my dependencies to their latest versions helps a bit, the content manager doesn't seem to crash anymore, but I still get errors in the console. The crash still occurs when I access the `strapi-plugin-menus` plugin, but you don't seem to use it, so I'm guessing it's not the root of the issue.

**Comment by abdbadgithub** [comment]
hi i updated the seo plugin from 1.8.0 to 1.9.1 (latest) and everything is now is working properly. maybe you need to update the menu plugin also in your case.

**Comment by Schlipak** [comment]
> hi i updated the seo plugin from 1.8.0 to 1.9.1 (latest) and everything is now is working properly. maybe you need to update the menu plugin also in your case.

I updated the SEO pugin and I still get the `Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'seoChecks')` error, though it doesn't crash the content manager.
The menu plugin is already at its latest version (1.4.1) though I might drop it since I can replicate what I need with single content-types and components.

**Comment by GregorSondermeier** [comment]
Try deleting your _node_modules_, folder, your _package-lock.json_/_yarn.lock_ files, then run `npm i`, `strapi build` and `strapi develop` again. Does that work?

**Comment by Renkas** [comment]
> Try deleting your _node_modules_, folder, your _package-lock.json_/_yarn.lock_ files, then run `npm i`, `strapi build` and `strapi develop` again. Does that work?

Tried that and the issue is still there

**Comment by Renkas** [comment]
Just to clarify. This problem seems to happen only when navigating in `Content Manager`. Rest of admin seems to work fine.

**Comment by cooperwfloyd** [comment]
Same behavior on my end, getting the follow error in my console when the page crashes.

```
Uncaught TypeError: N is not a function
    at _s (main.6d26c4cd.js:36:23092)
    at bl (main.6d26c4cd.js:36:24881)
    at Wd (main.6d26c4cd.js:36:42988)
    at gd (main.6d26c4cd.js:36:42517)
    at Vd (main.6d26c4cd.js:36:41540)
    at ll (main.6d26c4cd.js:36:38033)
    at V0 (main.6d26c4cd.js:34:3291)
    at main.6d26c4cd.js:36:35377
```

**Comment by zefman** [comment]
Same issue here. This seems to be a very common problem with Strapi, I have had this happen many times when trying to update.

**Comment by zefman** [comment]
A tip for anyone trying to figure out whats causing this. If you run the admin in watch mode rather than building it first you will get errors from the unminified code making it much easier to find whats actually going wrong. 

```
strapi develop --watch-admin
```

**Comment by Renkas** [comment]
This is what I get with --watch-admin flag:

```
react-dom.development.js:22932 Uncaught TypeError: destroy is not a function
    at safelyCallDestroy (react-dom.development.js:22932:1)
    at commitHookEffectListUnmount (react-dom.development.js:23100:1)
    at commitPassiveUnmountInsideDeletedTreeOnFiber (react-dom.development.js:25098:1)
    at commitPassiveUnmountEffectsInsideOfDeletedTree_begin (react-dom.development.js:25048:1)
    at commitPassiveUnmountEffects_begin (react-dom.development.js:24956:1)
    at commitPassiveUnmountEffects (react-dom.development.js:24941:1)
    at flushPassiveEffectsImpl (react-dom.development.js:27038:1)
    at flushPassiveEffects (react-dom.development.js:26984:1)
    at commitRootImpl (react-dom.development.js:26935:1)
    at commitRoot (react-dom.development.js:26682:1)
```

```
react-dom.development.js:18687 The above error occurred in the <CMEditViewExclude> component:

    at CMEditViewExclude (webpack-internal:///517988:21:96)
    at InjectionZone (webpack-internal:///151989:17:26)
    at aside
    at O (webpack-internal:///671893:30:21101)
    at div
    at O (webpack-internal:///671893:30:21101)
    at div
    at O (webpack-internal:///671893:30:21101)
    at div
    at O (webpack-internal:///671893:30:21101)
    at f (webpack-internal:///811276:17:16)
    at div
    at O (webpack-internal:///671893:30:21101)
    at d (webpack-internal:///649066:9:24)
    at main
    at O (webpack-internal:///671893:30:21101)
    at m (webpack-internal:///700185:16:23)
    at form
    at EditViewDataManagerProvider (webpack-internal:///623986:55:3)
    at SingleTypeFormWrapper (webpack-internal:///155634:38:34)
    at EditView (webpack-internal:///931652:73:29)
    at Permissions (webpack-internal:///403868:25:166)
    at EditViewLayoutManager (webpack-internal:///304664:29:34)
    at Route (webpack-internal:///616550:648:29)
    at Switch (webpack-internal:///616550:850:29)
    at SingleTypeRecursivePath (webpack-internal:///369784:33:10)
```

**Comment by PandaaAgency** [comment]
same issue here after upgrading to 4.11.1 : 
![image](https://github.com/strapi/strapi/assets/5455203/27234cec-59a7-4b94-9cf1-5596f63586df)

**Comment by boazpoolman** [comment]
This issue is likely caused by the [sitemap plugin](https://github.com/boazpoolman/strapi-plugin-sitemap).
A fix has since been released with `v2.0.11`. Upgrading to that version should do the trick.

The crash was caused by a faulty implementation of an async React `useEffect` hook.
See [the commit](https://github.com/boazpoolman/strapi-plugin-sitemap/commit/7208c87faf87b710007ecef49db00ba295bb7d65) that fixed the issue.

If you're not using the sitemap plugin - or upgrading doesn't fix the issue for you - you're probably using another plugin that also has this faulty `useEffect` implementatino in their codebase.

**Comment by Renkas** [comment]
This fixed it for me. Closing the issue.

**Comment by kratos-digital** [comment]
This is still the issue. I just installed Strapi using blog template and after I installed official SEO plugin I started to get white pages. There is no sitemap plugin or ant other plugin, other than official SEO.

It happens when I click on SEO in sidebar menu and then to any other link in sidebar, for example I go to SEO, then Media Library and I get white page.

No errors in terminal, even with --watch-admin flag.
